packages/substrate/src/fungible.ts

```diff
@@ -0,0 +1,158 @@
+import type {
+  Domainlike,
+  Environment,
+  SubstrateConfig,
+  SubstrateResource,
+} from '@buildwithsygma/core';
+import { LiquidityError, Config, FeeHandlerType, ResourceType } from '@buildwithsygma/core';
+import type { ApiPromise, SubmittableResult } from '@polkadot/api';
+import type { SubmittableExtrinsic } from '@polkadot/api-base/types';
+import { BN } from '@polkadot/util';
+
+import { BaseTransfer } from './base-transfer.js';
+import type { Fungible, SubstrateFee, Transfer } from './types.js';
+import {
+  deposit,
+  getBasicFee,
+  getFeeHandler,
+  getPercentageFee,
+  getLiquidity,
+} from './utils/index.js';
+
+export type SubstrateAssetTransferRequest = {
+  sourceDomain: Domainlike;
+  destinationDomain: Domainlike;
+  sourceNetworkProvider: ApiPromise;
+  resource: string | SubstrateResource;
+  amount: bigint;
+  destinationAddress: string;
+};
+
+async function checkDestinationFungibleHandler(
+  destinationDomain: SubstrateConfig,
+  resourceId: string,
+  amount: bigint,
+  sourceNetworkProvider: ApiPromise,
+): Promise<void> {
+  const handlerAddress = destinationDomain.handlers.find(
+    handler => handler.type === ResourceType.FUNGIBLE,
+  )?.address;
+  if (!handlerAddress) {
+    throw new Error('No Fungible handler configured on destination domain');
+  }
+
+  const destinationResource = destinationDomain.resources.find(
+    resource => resource.resourceId === resourceId,
+  );
+  const destinationHandlerBalance = await getLiquidity(
+    sourceNetworkProvider,
+    destinationResource as SubstrateResource,
+    handlerAddress,
+  );
+
+  if (destinationHandlerBalance < amount) {
+    throw new LiquidityError(destinationHandlerBalance);
+  }
+}
+
+export async function createSubstrateFungibleAssetTransfer(
+  transferRequestParams: SubstrateAssetTransferRequest,
+): Promise<SubstrateFungibleAssetTransfer> {
+  const config = new Config();
+  await config.init(process.env.SYGMA_ENV as Environment);
+
+  const transfer = new SubstrateFungibleAssetTransfer(transferRequestParams, config);
+
+  const destinationDomain = config.getDomainConfig(transfer.destinationAddress) as SubstrateConfig;
```

Comment:
Is it correct? `getDomainConfig` accepts chain id or caip id doesnt it

Reply:
yeah, it's a typo